atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampStore.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.nio.ByteBuffer;
+import java.util.ConcurrentModificationException;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.cassandra.thrift.Compression;
+import org.apache.cassandra.thrift.ConsistencyLevel;
+import org.apache.cassandra.thrift.CqlResult;
+import org.apache.cassandra.thrift.CqlRow;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.common.base.Throwables;
+import com.palantir.timestamp.DebugLogger;
+import com.palantir.util.debug.ThreadDumps;
+
+public class CassandraTimestampStore {
+    private static final Logger log = LoggerFactory.getLogger(CassandraTimestampStore.class);
+
+    private final CassandraKeyValueService cassandraKeyValueService;
+
+    public CassandraTimestampStore(CassandraKeyValueService cassandraKeyValueService) {
+        this.cassandraKeyValueService = cassandraKeyValueService;
+    }
+
+    /**
+     * Creates the timestamp table. Note that this operation is idempotent.
```

Comment:
note: we sometimes use `@Idempotent` to indicate the idempotence of the operation.